sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -293,13 +293,18 @@ private class JsonConverter : JsonConverter<DynamicData>
         {
             public override DynamicData Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
             {
+#if NET6_0_OR_GREATER
+                using JsonDocument document = JsonDocument.ParseValue(ref reader);
```

Comment:
Can this be disposed here? Doesn't DD need it?

Reply:
Are you talking about the DD constructor? I took a look at it and it simply stores the JsonElement in a field.

Reply:
No - the MutableJsonDocument constructor.  The new change removes the one that was calling JsonSerializer.Serialize() - we actually didn't need it, and it was a source of confusion.